devscripts/update-version.py

```diff
@@ -30,27 +32,41 @@ def get_new_version(revision):
 
 def get_git_head():
     with contextlib.suppress(Exception):
-        sp = subprocess.Popen(['git', 'rev-parse', '--short', 'HEAD'], stdout=subprocess.PIPE)
-        return sp.communicate()[0].decode().strip() or None
+        return subprocess.check_output(['git', 'rev-parse', 'HEAD'], text=True).strip() or None
 
 
-VERSION = get_new_version((sys.argv + [''])[1])
-GIT_HEAD = get_git_head()
-
-VERSION_FILE = f'''\
+VERSION_TEMPLATE = '''\
 # Autogenerated by devscripts/update-version.py
 
-__version__ = {VERSION!r}
+__version__ = {version!r}
 
-RELEASE_GIT_HEAD = {GIT_HEAD!r}
+RELEASE_GIT_HEAD = {git_head!r}
 
 VARIANT = None
 
 UPDATE_HINT = None
+
+CHANNEL = {channel!r}
```

Comment:
Leave VARIANT/UPDATE_HINT at bottom just in case someone's script is relying on it

Reply:
Only `VARIANT` and `UPDATE_HINT`? What about `__version__` or `RELEASE_GIT_HEAD`? What if they rely on `VARIANT` being on the 7th line from the top instead?